docs/8-veba-troubleshooting.md

```diff
@@ -0,0 +1,232 @@
+kube-system      weave-net-vs8ls                        2/2     Running              4          4d15h
+projectcontour   contour-5cddfc8f6-8hzd6                1/1     Running              1          4d15h
+projectcontour   contour-5cddfc8f6-jq7d8                1/1     Running              1          4d15h
+projectcontour   contour-certgen-f92l5                  0/1     Completed            0          4d15h
+projectcontour   envoy-gcmqt                            1/1     Running              1          4d15h
+vmware           tinywww-7fcfc6fb94-mfltm               1/1     Running              1          4d15h
+vmware           vmware-event-router-5dd9c8f858-5c9mh   0/1     CrashLoopBackoff     6          4d13h
+```
+
+One of the first things to look for is whether a pod is in a crash state. In this case, the vmware-event-router pod is crashing. We need to look at the logs with this command:
+
+```
+kubectl logs vmware-event-router-5dd9c8f858-5c9mh  -n vmware
+```
+
+Here is the command output:
+
+```
+ _    ____  ___                            ______                 __     ____              __
+| |  / /  |/  /      ______ _________     / ____/   _____  ____  / /_   / __ \____  __  __/ /____  _____
+| | / / /|_/ / | /| / / __  / ___/ _ \   / __/ | | / / _ \/ __ \/ __/  / /_/ / __ \/ / / / __/ _ \/ ___/
+| |/ / /  / /| |/ |/ / /_/ / /  /  __/  / /___ | |/ /  __/ / / / /_   / _, _/ /_/ / /_/ / /_/  __/ /
+|___/_/  /_/ |__/|__/\__,_/_/   \___/  /_____/ |___/\___/_/ /_/\__/  /_/ |_|\____/\__,_/\__/\___/_/
+
+
+[VMware Event Router] 2020/03/10 18:59:47 connecting to vCenter https://vc01.labad.int/sdk
+[VMware Event Router] 2020/03/10 18:59:52 could not connect to vCenter: could not create vCenter client: ServerFaultCode: Cannot complete login due to an incorrect user name or password.
+```
+
+The error message shows us that we made a mistake when we configured our username or password. We must now edit the Event Router JSON configuration file to fix the mistake.
+
+```
+vi /root/event-router-config.json
+
+```
+
+Here is some of the JSON from the config file - you can see the mistake in the credentials. Fix the credentials and save the file.
+```
+[{
+                "type": "stream",
+                "provider": "vmware_vcenter",
+                "address": "https://vc01.labad.int/sdk",
+                "auth": {
+                        "method": "user_password",
+                        "secret": {
+                                "username": "administrator@vsphere.local",
+                                "password": "WrongPassword"
+                        }
+                },
+                "options": {
+                        "insecure": "true"
+                }
+        },
+
+```
+
+We now fix the Kubernetes configuration with 3 commands - delete and recreate the secret file, then delete the broken pod. Kubernetes will automatically spin up a new pod with the new configuration
```

Comment:
Explain why this is needed. Users won't automatically understand that just fixing the local file is not enough. I.e. the file is used in Kubernetes and mounted into the event router pod as a secret. That's why we need to perform the following steps to update the secret for the event router pod...

Reply:
Added language as requested